### PR TITLE
add and migrate constants from `ctypes/__init__.pyi` to `_ctypes.pyi`

### PR DESCRIPTION
**Pull request by junkmd**
I have moved some constants in `ctypes/__init__.pyi` to `_ctypes.pyi` as defined in the implementation.

I also added some constants defined in `_ctypes`.

I will move other stuffs eventually, but I will try to do it little by little.

- related to #8571 and #8633

### CODE REVIEW

**Review by AlexWaygood — APPROVED** [review]
Great stuff, thank you!

**Comment by AlexWaygood** [comment]
Hmm, I'm not sure what's up with the pytype crash. This _might_ be as a result of the circular import :(

I'll try bumping the pytype version in CI to see if that helps, since I think pytype's done some work on improving their name-resolution logic recently.

**Comment by AlexWaygood** [comment]
> I'll try bumping the pytype version in CI to see if that helps, since I think pytype's done some work on improving their name-resolution logic recently.

No, looks like that didn't fix it ☹️ (cc. @rchen152)

As a workaround for now, maybe you could leave `RTLD_LOCAL` and `RTLD_GLOBAL` as they are in `ctypes/__init__.pyi`, and add a TODO comment saying that ideally we would re-export them from `_ctypes.pyi`?

**Comment by Akuli** [comment]
We could also use `tests/pytype_exclude_list.txt`.

**Comment by junkmd** [comment]
@Akuli 
@AlexWaygood 

It is curious that no other modules using circular imports have this error, such as `importlib`.

I will add workarounds to `tests/pytype_exclude_list.txt` in the hope that someone will do a fix for `pytype`.

**Comment by Akuli** [comment]
Circular imports don't automatically mean that pytype will crash. They only crash pytype when hitting some corner case that wasn't considered before.

**Comment by junkmd** [comment]
@Akuli 
(cc: @AlexWaygood)

I added all the modules in `stdlib/ctypes` to the list and still got the error.

According to the error messages, some errors seem to occur in various modules importing `ctypes`.

Do I have to change the `pyi` file side currently?

**Comment by Akuli** [comment]
Because `_typeshed/__init__.pyi` imports `ctypes`, and because lots of other modules use `_typeshed`, using the exclude list would mean we need to exclude a lot of things. It isn't a good idea after all.

Instead, we can:
- Wait for pytype to get fixed
- Try reverting some of the changes in this PR to make pytype pass (and fix them in a future PR once the pytype bug is fixed)

File: stdlib/ctypes/__init__.pyi [hunk 1]
```diff
@@ -5,15 +5,15 @@ from collections.abc import Callable, Iterable, Iterator, Mapping, Sequence
 from typing import Any, ClassVar, Generic, TypeVar, Union as _UnionT, overload
 from typing_extensions import TypeAlias
 
+from _ctypes import RTLD_GLOBAL as RTLD_GLOBAL, RTLD_LOCAL as RTLD_LOCAL
```
Comment (Akuli) [inline stdlib/ctypes/__init__.pyi:8]:
```suggestion
# TODO: import these from _ctypes once it no longer breaks pytype
RTLD_GLOBAL: int
RTLD_LOCAL: int
```

File: tests/pytype_exclude_list.txt [hunk 2]
```diff
@@ -10,6 +10,12 @@ stubs/mysqlclient/MySQLdb/__init__.pyi
 stubs/mysqlclient/MySQLdb/connections.pyi
 stubs/mysqlclient/MySQLdb/cursors.pyi
 
+# errors about circular imports
+stdlib/ctypes/__init__.pyi
+stdlib/ctypes/util.pyi
+stdlib/ctypes/wintypes.pyi
+stdlib/_ctypes.pyi
+
```
Comment (Akuli) [inline tests/pytype_exclude_list.txt:18]:
```suggestion
```

**Comment by junkmd** [comment]
@Akuli 
Thank you for code correction.

**Review by AlexWaygood — APPROVED** [review]
LGTM!

**Pull request merged**